### DISCUSSION TITLE
Privacy with Live Context - Clarification

### DISCUSSION THREAD

**Discussion by fregys** [pieces-app/support#390] [Q&A] [answered]
Hi, I have just started using Pieces for Developers and it looks very cool! I have read the documentation page but still have questions about its privacy.

When using Live Context with Workstream Pattern Engine (WPE):
1. If I use Enrichment Mode: **Blended** - My snippets captured by WPE are sent to cloud AI to get enriched, meaning my personal data leaves my computer and is sent to the cloud AI provider I choose, correct?
2. If I use Enrichment Mode: **Local** - My snippets are not sent anywhere, right?

Then when actually prompting while using Live Context:
1. If I use **cloud AI**, your page says all relevant context is sent. What is usually meant by relevant data? I am guessing it is those screenshots or snippets Pieces (WPE) gathered in the background?
2. If I use **local** no data is sent to any external services?

If I understand it correctly, then when I use WPE with **local** and Live Context prompting with **local LLM**, my data are not shared anywhere on the internet through Pieces software, correct? Not even **Pieces** third-party services or **Pieces** itself?

**Comment by rosie-at-pieces** [comment, marked as answer]
Hi @FilipKaspar good questions and so sorry about the delay in answering here. A few clarifications:

> If I use Enrichment Mode: Blended - My snippets captured by WPE are sent to cloud AI to get enriched, meaning my personal data leaves my computer and is sent to the cloud AI provider I choose, correct?

The WPE doesn't explicitly capture snippets - you have to explicitly save snippets using our save function. When you save a snippet and you use blended processing, we do leverage some cloud processing for a few capabilities like link extraction and suggested save, but if you save a snippet and use local processing, we do not leverage that and all processing happens entirely on-device. 

> If I use cloud AI, your page says all relevant context is sent. What is usually meant by relevant data? I am guessing it is those screenshots or snippets Pieces (WPE) gathered in the background?

We do not send (or even save) any screenshots when we capture information using the WPE; data we capture is immediately converted into a text format and saved on-device and the original data capture is discarded. When we say relevant data is sent, we mean data in that text format that is deemed relevant by the prompt; we use RAG for this. 

> If I use local no data is sent to any external services?

Correct! This is entirely on-device AI ✨ 

> If I understand it correctly, then when I use WPE with local and Live Context prompting with local LLM, my data are not shared anywhere on the internet through Pieces software, correct? Not even Pieces third-party services or Pieces itself?

Correct! Pieces can never see your snippets or copilot conversation in any circumstances; it is entirely saved on-device.

**Reply by fregys** [reply]
Awesome, thank you for the clarification! 🙂

**Reply by rosie-at-pieces** [reply]
Just a note for future readers: we have rebranded the WPE (Workstream Pattern Engine) to the (LTM) Long-Term Memory Engine, and Live Context is now Long-Term Memory
